rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/expression/ExpressionGenerator.xtend

```diff
@@ -1004,7 +996,7 @@ class ExpressionGenerator extends RosettaExpressionSwitch<JavaStatementBuilder,
 	}
 
 	override protected caseSumOperation(SumOperation expr, Context context) {
-		val itemType = typeProvider.getRType(expr.argument).toJavaReferenceType
+		val itemType = typeProvider.getRMetaAnnotatedType(expr.argument).toJavaReferenceType
 		buildListOperationNoBody(expr, "sum" + itemType.simpleName, MAPPER_C.wrapExtends(itemType), [MAPPER_S.wrap(itemType)], context.scope)
```

Comment:
This breaks e.g., when taking the sum of integers with metadata.
```
func MyFunc:
  inputs:
    n int (0..*)
      [metadata scheme]
  output:
    result int (1..1)
  set result:
    n sum
```

Reply:
fixed